frontend/src/components/UpgradeImage.tsx

```diff
@@ -5,11 +5,24 @@ interface IUpgradeImageProps {
 }
 
 export default function UpgradeImage({ picture }: IUpgradeImageProps) {
+  // imageElements is currently type "any" because TS had issues assigning type "Element[]" to the type "string | Element[]"
+  let imageElements: any = undefined;
   switch (picture.type) {
     case ImageType.string:
-      return picture.image;
+      imageElements = picture.imageArr.join("");
```

Comment:
This is a personal preference, but I think `.toString()` rather than `.join("")` makes more sense at a glance. `.join()` kinda implies that it is working on multiple items when it's really only working on one. This comment is assuming that any string image is always going to be just a single item in the array though. If there are plans to have multiple items for some reason then you can ignore this.